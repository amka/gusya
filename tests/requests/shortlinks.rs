use gusya::app::App;
use insta::assert_debug_snapshot;
use loco_rs::testing::prelude::*;
use serial_test::serial;

#[tokio::test]
#[serial]
async fn can_get_redirect() {
    request::<App, _, _>(|request, _ctx| async move {
        let res = request.get("/short_link").await;
        assert_eq!(res.status_code(), 302);
        assert_eq!(res.header("location"), "https://example.com");
    })
    .await;
}

#[tokio::test]
#[serial]
async fn can_create_short_link() {
    request::<App, _, _>(|request, _ctx| async move {
        let res = request
            .post("/")
            .json(&serde_json::json!({
                "url": "https://example.com"
            }))
            .await;

        assert_debug_snapshot!((res.status_code(), res.text()));
    })
    .await;
}
