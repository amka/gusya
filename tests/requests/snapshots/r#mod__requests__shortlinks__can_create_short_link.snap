---
source: tests/requests/shortlinks.rs
expression: "(res.status_code(), res.text())"
---
(
    200,
    "{\"short_code\":\"short_code\"}",
)
